pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -302,6 +302,7 @@ func (s *Scheduler) add(ctx context.Context, pod *corev1.Pod) error {
 	// Pick existing node that we are about to create
 	for _, nodeClaim := range s.newNodeClaims {
 		if err := nodeClaim.Add(pod, s.cachedPodData[pod.UID]); err == nil {
+			s.cluster.MarkPodToNodePoolSchedulingDecision(pod, nodeClaim.Labels[v1.NodePoolLabelKey])
```

Comment:
How expensive is it to do one iteration through all the pods at the end? You could just iterate through the results and mark pod scheduling decisions with the NodePool attached in the same place I think rather than having to create an internal store to capture this

Reply:
Made an update